hedera-node/src/main/java/com/hedera/services/txns/validation/TransferListChecks.java

```diff
@@ -47,10 +44,17 @@ public static boolean isNetZeroAdjustment(TransferListOrBuilder wrapper) {
 	}
 
 	public static boolean hasRepeatedAccount(TransferList wrapper) {
-		var unique = new HashSet<AccountID>();
-		for (AccountAmount adjustment : wrapper.getAccountAmountsList()) {
-			unique.add(adjustment.getAccountID());
+		final int n = wrapper.getAccountAmountsCount();
+		if (n < 2) {
+			return false;
 		}
-		return unique.size() < wrapper.getAccountAmountsCount();
+		for (var i = 0; i < n - 1; i++) {
+			for (var j = i + 1; j < n; j++) {
+				if (wrapper.getAccountAmounts(i).getAccountID().equals(wrapper.getAccountAmounts(j).getAccountID())) {
+					return true;
+				}
+			}
+		}
+		return false;
```

Comment:
The original implementation was faster, the new implementation could also utilize `HashSet` to make it O(N) instead of O(N^2)

Reply:
`N <= 10` here. 😁

Reply:
I see, thanks!